pkgs/games/polymc/default.nix

```diff
@@ -0,0 +1,71 @@
+{ lib, mkDerivation, makeDesktopItem, fetchFromGitHub, cmake, jdk8, jdk, zlib, file, makeWrapper, xorg, libpulseaudio, qtbase, libGL, msaClientID ? "" }:
+
+let
+  libpath = with xorg; lib.makeLibraryPath [ libX11 libXext libXcursor libXrandr libXxf86vm libpulseaudio libGL ];
+in
+mkDerivation rec {
+  pname = "polymc";
+  version = "1.0.4";
+
+  src = fetchFromGitHub {
+    owner = "PolyMC";
+    repo = "PolyMC";
+    rev = version;
+    sha256 = "sha256-8aya0KfV9F+i2qBpweWcR9hwyTSQkqn2wHdtkCEeNvk=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ cmake file makeWrapper ];
+  buildInputs = [ qtbase jdk8 zlib ];
+
+  patches = [ ./0001-pick-latest-java-first.patch ];
+
+  postPatch = ''
+    # hardcode jdk paths
+    substituteInPlace launcher/java/JavaUtils.cpp \
+      --replace 'scanJavaDir("/usr/lib/jvm")' 'javas.append("${jdk}/lib/openjdk/bin/java")' \
+      --replace 'scanJavaDir("/usr/lib32/jvm")' 'javas.append("${jdk8}/lib/openjdk/bin/java")'
+
+  '' + lib.optionalString (msaClientID != "") ''
+    # add client ID
+    substituteInPlace CMakeLists.txt \
+      --replace '17b47edd-c884-4997-926d-9e7f9a6b4647' '${msaClientID}'
+  '';
+
+  cmakeFlags = [ "-DLauncher_LAYOUT=lin-system" ];
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "polymc";
+      desktopName = "PolyMC";
+      genericName = "Minecraft Launcher";
+      comment = "Free, open source launcher and instance manager for Minecraft.";
+      icon = "launcher";
+      exec = "polymc";
+      categories = "Game";
+      terminal = "false";
+    })
+  ];
+
+  postInstall = ''
```

Comment:
It is better to put the `let libpath =... ; in ` here, nearer to its intended use.

Reply:
Made it more readable